src/main/java/org/apereo/openequella/tools/toolbox/utils/CheckFilesUtils.java

```diff
@@ -0,0 +1,226 @@
+			for (String s : findOnlyInFirst(keys1, keys2)) {
+				onlyInFirst.add(rows1.get(s));
+			}
+			List<ResultsRow> onlyInSecond = new ArrayList<ResultsRow>();
+			for (String s : findOnlyInFirst(keys2, keys1)) {
+				onlyInSecond.add(rows2.get(s));
+			}
+			return new ResultComparison(r1.getName(), onlyInFirst,
+							r2.getName(), onlyInSecond);
+		} catch (Exception e) {
+			logger.error("Error trying to compare results:  {}",
+							e.getMessage(), e);
+			return null;
+		}
+	}
+
+	//Only returns the n-1 file if n is greater than 1.
+	// It doesn't return the latest file since that just occured.
+	public static File findPreviousErrorStats() {
+		File dir = new File(Config.get(Config.CF_OUTPUT_FOLDER));
+		String[] errorStats = dir.list(new OnlyErrorStatsFileFilter());
+
+		if(errorStats == null) {
+			return null;
+		}
+
+		List<String> errorStatsArray = Arrays.asList(errorStats);
+		Collections.sort(errorStatsArray);
+		File f = null;
+		if (errorStatsArray.size() > 1) {
+			f = new File(Config.get(Config.CF_OUTPUT_FOLDER),
+					errorStatsArray.get(errorStatsArray.size() - 2));
+		}
+		return f;
+	}
+
+	public static Collection<String> findOnlyInFirst(Set<String> s1,
+																									 Set<String> s2) {
+		Set<String> base = new HashSet<String>();
+		base.addAll(s1);
+		base.removeAll(s2);
+		return base;
+	}
+
+	private static Map<String, ResultsRow> slurpAndNormalizeCsvFile(File f)
+			throws Exception {
+		Map<String, ResultsRow> rows = new HashMap<String, ResultsRow>();
+		if (!f.exists()) {
+			return rows;
+		}
+		BufferedReader br = new BufferedReader(new FileReader(f));
+		String row = br.readLine();
+		while (row != null) {
+			if (row.startsWith("Institution") || row.startsWith("CheckFiles type")
+					|| row.startsWith("Report Timestamp")
+					|| row.startsWith("Report Duration")
+					|| row.startsWith("Database URL")
+					|| row.startsWith("Database Type")
+					|| row.startsWith("Filestore attachments")
+					|| row.startsWith("Filter By Institution")
+					|| row.startsWith("Filter By Collection")
+					|| row.startsWith("# Of") || row.startsWith("Item UUID")
+					|| row.startsWith("Stats")
+					|| row.startsWith("Report Start")
+					|| row.startsWith("Report End")
+					|| row.startsWith("# Of queries ran")
+					|| row.startsWith("Total duration of queries (ms)")
+					|| row.startsWith("Average duration of queries (ms)")) {
+				// Ignore - not a 'results row'.
+			} else {
+				ResultsRow parsedRow = slurpAndNormalizeCsvRow(row);
+				rows.put(parsedRow.getPrimaryKey(), parsedRow);
+			}
+			row = br.readLine();
+		}
+		br.close();
+		return rows;
+	}
+
+	/**
+	 * Lines parsed into: Item UUID Item Version ItemStatus Collection UUID
+	 * Attachment Type Attachment UUID Attachment Status Attachment Response
+	 * Code Item Name *** Attachment Filepath ***
+	 *
+	 * *** Quoted CSV data. not used in comparisons, thus not set.
+	 *
+	 * @param row
+	 * @returnt
+	public void testFindOnlyInFirst() {
```

Comment:
Is this function intended to be commented out?

Reply:
thanks for the catch, I'll remove it.